minvws/nl-kat-coordination
# fix: toggle styling

### Changes

- Fixes toggle button styling and updates to the new designs. 
- Adds focus and hover state. 

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/2495 

### Demo

Before: 
![Screenshot 2024-09-03 at 18 11 52](https://github.com/user-attachments/assets/a65a973d-d18f-4e1b-9266-b7c971d0676c)

After:
<img width="1276" alt="Screenshot 2024-09-03 at 19 17 25" src="https://github.com/user-attachments/assets/c9238c3e-5884-4dc9-8e19-7fb08b45eb73">

### To do

- [x] Look into which ARIA attributes are needed (if any) to signal the current state.
- [x] Style based on ARIA attribute instead of "selected" class. (selected class can be removed after).
- [x] Current border-radius isn't in our border-radii set. Decide if we want to expand our set, or change the border radius of this element. 
- [x] Update border-radii set or border-radius.

### QA notes
This component can be tested in the KAT-alogus. 

Please check: 
- If the states are behaving as expected. (focus, hover, selected, active) 
- If any other unexpected behaviour occur
- If the toggle is implemented as designed (https://www.figma.com/design/MDbpc9K0qjyaoRGAGiDX6t/%F0%9F%8E%A8-KAT-%7C-Design-System-%7C-Components?m=auto&node-id=2066-1144&t=CjfLtkVzzAbmMsQ1-1)

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
